test/testenv/s3utils.go

```diff
@@ -226,3 +229,74 @@ func UploadFilesToS3(testS3Bucket string, s3TestDir string, applist []string, do
 	}
 	return uploadedFiles, nil
 }
+
+// DisableAppsToS3 download apps from S3, untar them, modify their conf file to disable them, re-tar and upload the disabled version to S3
+func DisableAppsToS3(testDataS3Bucket string, s3AppDir string, downloadDir string, appFileList []string, s3TestDir string) ([]string, error) {
+
+	// Download apps tar files
+	DownloadFilesFromS3(testDataS3Bucket, s3AppDir, downloadDir, appFileList)
+
+	// Create a folder named 'untarred_apps' to store untarred apps folders
+	untarredAppsMainFolder := downloadDir + "/untarred_apps"
+	cmd := exec.Command("mkdir", untarredAppsMainFolder)
+	cmd.Run()
+
+	// Create a folder named 'disabled_apps' to stored disabled apps tgz files
+	disabledAppsFolder := downloadDir + "/disabled_apps"
+	cmd = exec.Command("mkdir", disabledAppsFolder)
+	cmd.Run()
+
+	for _, key := range appFileList {
+		// Create a specific folder for each app in 'untarred_apps'
+		tarfile := downloadDir + "/" + key
+		lastInd := strings.LastIndex(key, ".")
+		untarredCurrentAppFolder := untarredAppsMainFolder + "/" + key[:lastInd]
+		cmd := exec.Command("mkdir", untarredCurrentAppFolder)
+		cmd.Run()
+
+		// Untar the app
+		cmd = exec.Command("tar", "-xf", tarfile, "-C", untarredCurrentAppFolder)
```

Comment:
In my opinion it would be better to use go library to untar like here https://medium.com/@skdomino/taring-untaring-files-in-go-6b07cf56bc07
Depending on exec package installed on OS might raise issues in future.

Reply:
As discussed, please open another task to create util APIs for tar and untar

Reply:
Will use CSPL-1964 to address this